test_e2e.sh

```diff
@@ -194,6 +194,48 @@ fi
 RUNID="${RANDOM}${RANDOM}"
 DIR="/tmp/git-sync-e2e.$RUNID"
 mkdir "$DIR"
+function final_cleanup() {
+  if [[ "${CLEANUP:-}" == 0 ]]; then
+      echo "leaving logs in $DIR"
+  else
+      rm -rf "$DIR"
+  fi
+}
+# Set the trap to call the final_cleanup function on exit.
+trap final_cleanup EXIT
+
+skip_github_app_test="${SKIP_GITHUB_APP_TEST:-false}"
```

Comment:
nit: I think skipping may be a more user friendly default, since in most cases a developer won't have github app credentials ready to use

Reply:
That makes sense as long as the github action workflow doesn't skip the github app test.